### ISSUE TITLE
[Component] Sortable table

### ISSUE THREAD

**Issue by shawnbot** [uswds/uswds#1643]
As mentioned in #1079, it would be really nice to offer a sortable table feature, or at least styles that indicate the sorted status of a table's column(s). To aid accessibility, we should offer the [`aria-sort` attribute](https://www.w3.org/TR/wai-aria/states_and_properties#aria-sort) as the styling mechanism. Specifically, we can probably do _all_ of the styling in CSS with generated content along the lines of:

```scss
[usa-sort] {
  background-color: $sortable-background;
  position: relative;

  &[aria-sort] {
    background-color: $sortable-background-sorted;

    &:after {
      position: absolute;
    }
  }

  &[aria-sort=ascending]:after {
    content: "▼";
  }

  &[aria-sort=descending]:after {
    content: "▲";
  }
}

```

As in all things a11y, we have to be super conscious of contrast and ensure that the sorted column header is sufficiently distinguishable from the others.

Functionally, sortable tables can be tricky to implement because we have to accommodate different types of sortable data: namely, that numbers should be sorted numerically and strings alphanumerically. Ideally, there would be a way to control whether each column is sortable and how it should be sorted. So maybe sorting is enabled via an optional attribute on the header, e.g. `usa-sort="num"` to enable numeric sorting or `usa-sort="alpha"` for alphanumeric.

And, because numbers can be formatted in all sorts of ways that are hard to parse as numbers for the sake of sorting in JavaScript (e.g. large negative values like `$(1,000,000)`), we should provide a way to pass a "raw" representation of the sortable value for each cell:

```html
<table usa-sortable="true">
  <thead>
    <tr>
      <th usa-sort="alpha">Product</th>
      <th usa-sort="num" aria-sort="ascending">Price</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Widget B</td>
      <td usa-sort-value="1234.56">$1,234.56</td>
    </tr>
    <tr>
      <td>Widget A</td>
      <td usa-sort-value="2016.33">$2,016.33</td>
    </tr>
  </tbody>
</table>
```

All of the behavior should be implemented with event delegation and keep state in DOM attributes so that tables can be modified by other code and sorting updated via user interactions.

**Comment by thisisdano** [comment]
Thanks for contributing to USWDS. Unfortunately, this issue is not on our current [product roadmap](https://designsystem.digital.gov/whats-new/product-roadmap/).

**Comment by sknep** [comment]
Sortable tables are in progress here and the implementation is quite similar to your recommendations above: #3950
